src/Graphing/Enrich.hs

```diff
@@ -0,0 +1,230 @@
+
+-- | Transform all 'Conan' dependency to 'Archive' dependency.
+-- It performs cli-side license scanning and upload of results. In this,
+-- transformation,
+-- -
+-- * It preserves edges of conan dependencies in the transformation.
+-- * TODO: Ignore non-production dependency in transformation
+conanToArchives ::
+  ( Has Diagnostics sig m
+  , Has (Lift IO) sig m
+  , Has StickyLogger sig m
+  , Has Logger sig m
+  , Has Exec sig m
+  , Has ReadFS sig m
+  , Has FossaApiClient sig m
+  ) =>
+  Path Abs Dir -> -- directory of the project's manifest
+  Graphing Dependency -> -- graph whose conan dependency to make archives
+  m (Graphing Dependency) -- graph with archive dependencies
+conanToArchives rootPath g =
+  -- If we do not have any conan dependencies, we do not
+  -- need to do any work!
+  case (null unableToTransformConanDep, transformedVendorDep) of
+    (True, Nothing) -> pure g
+    (False, _) -> fatal $ FailedToTransformConanDependency unableToTransformConanDep
+    (True, Just depsAndVendorDeps) -> do
+      let vendorDeps = NE.map snd depsAndVendorDeps
+
+      -- 1. We upload all vendor deps (transformed from conan deps),
+      -- since conan dependencies have to be scanned
+      -- before they can be referenced in graph as archive deps!
+      archiveLocators <-
+        licenseScanSourceUnit
+          -- always perform scan as vendor deps's
+          -- manifest are rarely updated with changes in practice
+          SkippingNotSupported
+          Nothing -- Ignore path filters
+          (FullFileUploads False)
+          rootPath
+          vendorDeps
+
+      -- 2. We make archive archive dependencies from locators, and original
+      -- conan dependencies.
+      let (failed, archiveDeps) =
+            partitionEithers $
+              map (locatorToArchiveDep allConanDeps) $
+                toList archiveLocators
+
+      unless (null failed) $
+        fatal $
+          FailedToTransformLocators failed
+
+      -- 3. We replace all conan dependencies with archive dependencies from
+      -- original graph. If we are unable to find twin of archive dep
+      -- (e.g. sourcing conan dep), we fail fatally!
+      case fromList <$> archiveToConanDep archiveDeps of
+        Left lonelyArchiveDeps -> fatal $ UnableToFindTwinOfArchiveDep lonelyArchiveDeps
+        Right registry -> pure $ gmap (\graphDep -> Map.findWithDefault graphDep graphDep registry) g
+  where
+    allDeps :: [Dependency]
+    allDeps = vertexList g
+
+    allConanDeps :: [Dependency]
+    allConanDeps = filter isConanDep allDeps
+
+    unableToTransformConanDep :: [Dependency]
+    unableToTransformConanDep = fst $ toVendorDeps allConanDeps
+
+    transformedVendorDep :: Maybe (NonEmpty (Dependency, VendoredDependency))
+    transformedVendorDep = nonEmpty . snd $ toVendorDeps allConanDeps
+
+    archiveToConanDep :: [Dependency] -> Either [Dependency] [(Dependency, Dependency)]
```

Comment:
[Optional] My understanding here is that the `Left` side of the `Either` is a list of deps with no partner. If that's true I'd recommend making a small newtype to describe the difference:

```haskell
newtype LonelyDeps = LonelyDeps  { unLonelyDeps :: [Dependency] }
  deriving (Eq, Ord)
  deriving newtype (Show) 
```

I think this replacement should only need a few calls to the constructor and `unLonelyDeps` in this module to work.